lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -140,13 +141,14 @@ sub list_running_ajax {
     my @running;
     while (my $job = $running->next) {
         my $job_id = $job->id;
+        my $test_name = $job->settings_hash->{JOB_TEMPLATE_NAME} ? $job->settings_hash->{JOB_TEMPLATE_NAME} : $job->TEST;
```

Comment:
Seems like this line - as well as the similar subsequent lines - should be a member function of the job result.